src/utils/AuthProvider.tsx

```diff
@@ -0,0 +1,116 @@
+'use client';
+
+import { AuthError, AuthResponse, Session } from '@supabase/supabase-js';
+import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { UUID } from 'crypto';
+import supabase from '@/api/supabase/createClient';
```

Comment:
small nit: we can use `import type` for imports from `@supabase/supabase-js` and `crypto`, since we'll only need the type declarations